### PR TITLE
compress and angle

### LINKED ISSUE #12930: angle

### LINKED ISSUE #12933: compress

### PR DESCRIPTION
**Pull request by AkshayC2001**
Close #12930  
Close #12933

### CODE REVIEW

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -230,6 +230,12 @@ def diagonal(self, *, offset=0, axis1=0, axis2=1):
             axis2=axis2,
         )
 
+    def compress(self, value, *, axis=None):
+        return np_frontend.compress(
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:234]: This function is isn't yet implemented in the numpy frontend
Need to implement it in the frontend side before implementing it's instance method

Comment (AkshayC2001) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py]: i removed compress function and kept only angle. correct me if i am wrong.
do i have to wait for ivy to add compress in frontend or do i have to add compress in frontend

Comment (hirwa-nshuti) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py]: Only the function needs to be implemented in the numpy frontend.
If it is not in ivy functional API you can use composition of ivy functions to implement behavior of this function

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @AkshayC2001 

The tests are failing for `angle`
Would you also add the issue number of angle to this PR so it can be linked.
Thanks

File: ivy/functional/frontends/numpy/ndarray/ndarray.py (hunk 1 above)
Comment (AkshayC2001) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py]: Ahhh now I understood that thank you

**Comment by AkshayC2001** [comment]
Hello, @hirwa-nshuti can you check the code please

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -1293,6 +1293,38 @@ def test_torch_instance_amin(
         on_device=on_device,
     )
 
+# angle
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="angle",
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric", full=True),
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:1302]: the function works with floats

File: ivy/functional/frontends/numpy/manipulation_routines/changing_array_shape.py [hunk 3]
```diff
@@ -9,6 +9,9 @@
 def reshape(x, /, newshape, order="C"):
     return ivy.reshape(x, shape=newshape, order=order)
 
+@to_ivy_arrays_and_back
+def compress(condition, x, axis=None, out=None):
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/manipulation_routines/changing_array_shape.py:13]: ivy functional API doesn't have this function you will need to implement it using composition of other ivy functions

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @AkshayC2001 

Check the comments below😊

File: ivy/functional/frontends/numpy/manipulation_routines/changing_array_shape.py (hunk 3 above)
Comment (AkshayC2001) [reply ivy/functional/frontends/numpy/manipulation_routines/changing_array_shape.py:13]: there is no extract function to mask the array while compressing. do i have to implement extract function too?

**Comment by AkshayC2001** [comment]
> Hey @AkshayC2001 
> 
> Check the comments below😊

I have implemented the compress function. @hirwa-nshuti can you check the code

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 4]
```diff
@@ -1201,6 +1200,48 @@ def test_numpy_instance_add__(
...
+    init_flags,
+    method_flags,
+    frontend,
+    on_device,
+):
+    input_dtypes_bool, x = dtype_and_x
+    input_dtypes_numeric, y = dtype_and_y
+
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtypes,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:1230]: Here you passed wrong dtypes argument

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
